src/mono/mono/mini/aot-runtime.c

```diff
@@ -2443,13 +2443,16 @@ load_container_amodule (MonoAssemblyLoadContext *alc)
 	if (!container_assm_name || container_amodule)
 		return;
 
+	char *local_ref = container_assm_name;
+	container_assm_name = NULL;
```

Comment:
Without locking, there is still a race here.

Reply:
Would `mono_aot_lock` help?

Reply:
It should use mono_loader_lock (), thats our global lock.